profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.admin.inc

```diff
@@ -24,6 +24,67 @@ function nexteuropa_varnish_admin_settings_form($form, $form_state) {
     '#default_value' => variable_get('nexteuropa_varnish_default_purge_rule', FALSE),
   );
 
+  $form = system_settings_form($form);
+  array_unshift($form['#submit'], '_nexteuropa_varnish_default_rule_state_check');
+
   return system_settings_form($form);
 
 }
+
+/**
+ * Implements hook_FORM_ID_form_validate().
+ */
+function nexteuropa_varnish_admin_settings_form_validate($form, &$form_state) {
+  $rule_state = $form_state['values']['nexteuropa_varnish_default_purge_rule'];
+  if ($rule_state && _nexteuropa_varnish_check_node_rules()) {
+    form_set_error(
+      'settings',
+      t('You can not enable the default purge rule while "Purge rules" type of node exist.')
+    );
+  }
+}
+
+/**
+ * Helper submit function for checking the PIWIK advanced rules state.
+ */
+function _nexteuropa_varnish_default_rule_state_check($form, &$form_state) {
+  // Checking the state of the default purge rule checkbox.
+  $rule_state = $form_state['values']['nexteuropa_varnish_default_purge_rule'];
+  $rule_state_var = variable_get('nexteuropa_varnish_default_purge_rule', FALSE);
+
+  switch ($rule_state) {
+    case 0:
+      $state = t('off');
+      break;
+
+    case 1:
+      $state = t('on');
+      break;
+  }
+
+  if ($rule_state != $rule_state_var) {
+    variable_set('nexteuropa_varnish_default_purge_rule', $rule_state);
```

Comment:
It is already done by the default admin submit, why do we have to set the variable again?